cpp/src/parquet/file_writer.cc

```diff
@@ -27,15 +27,33 @@
 #include "parquet/exception.h"
 #include "parquet/platform.h"
 #include "parquet/schema.h"
+#include "parquet/types.h"
 
 using arrow::MemoryPool;
 
 using parquet::schema::GroupNode;
 
 namespace parquet {
 
+namespace {
+
+void DetermineCompressionCodecAndLevel(const WriterProperties* properties,
+                                       const std::shared_ptr<schema::ColumnPath>& path,
+                                       Compression::type& outCompressionCodec,
+                                       int32_t& outCompressionLevel) {
```

Comment:
Does not conform to style guide

* Use pointers for "out" parameters
* Use `snake_case` for variable names